test/fail_compilation/fail_shared.d

```diff
@@ -0,0 +1,25 @@
+/*
+//REQUIRED_ARGS:-restrictiveshared
+TEST_OUTPUT:
+---
+fail_compilation/fail_shared.d(14): Error: Trying to accsess shared state x
+fail_compilation/fail_shared.d(15): Error: Trying to accsess shared state x
+fail_compilation/fail_shared.d(16): Error: Trying to accsess shared state x
+fail_compilation/fail_shared.d(22): Error: Trying to accsess shared state x
+---
+*/
+
+
+int f1(shared int x)
+{
+    x += 7; // Error: Trying to accsess shared state x
+    x = 22; // Error: Trying to accsess shared state x
+    return x; // Error: Trying to accsess shared state x
+
+}
+
+int sync(shared int *x, int y)
+{
+    return x = y; // Error: Trying to accsess shared state x
```

Comment:
`x` is a pointer... assigning int to pointer?

Reply:
auto deref.

Reply:
auto deref on assignment? is that really a thing?